internal/inputs/netdial.go

```diff
@@ -60,7 +63,7 @@ func NetDialWithTimeout(dataStore *[]interface{}, command load.Command, dataSamp
 			}
 		}
 		c <- struct{}{}
```

Comment:
In case the context is stoped, I think that this go routine will "leak".
I was thinking that maybe instead of those changes, making the chan `c` a buffered chan of length 1, will fix the issue with dialConn not being closed. WDYT?

Reply:
@cristianciutea thanks for pointing out the go routine "leak".  I tried buffered chan of length 1. It did not fix the dialConnect not being closed issue. 

I made a new update to avoid the go routine "leak" issue. Please have a look.  ta

Reply:
I've just saw how we call ReadLine(), you're right using a buffered channel will not fix the issue.
The latest commit will fix the go routine leak. Thank you!